File: api/models/manga.go
package models

import (
	"time"
)

type Manga struct {
	ID              uint       `gorm:"primaryKey" json:"id"`
	Titulo          string     `gorm:"type:varchar(120);not null" json:"titulo"`
	Sinopse         string     `gorm:"type:text;not null" json:"sinopse"`
	Imagem          string     `gorm:"type:varchar(150);not null" json:"imagem"`
	DataCriacao     *time.Time `gorm:"type:date" json:"data_criacao,omitempty"`
	DataFinalizacao *time.Time `gorm:"type:date" json:"data_finalizacao,omitempty"`
	EditoraID       uint       `gorm:"not null" json:"editora_id"`
	Editora         *Editora   `json:"editora,omitempty"`
	Generos         []Genero   `gorm:"many2many:manga_generos;" json:"generos"`
	Autores         []Autor    `gorm:"many2many:manga_autores;" json:"autores"`
	Capitulos       []Capitulo `json:"capitulos"`
}

File: app/lib/models/manga.dart
import 'package:app/models/capitulo.dart';

class Manga {
  final int id;
  final String titulo;
  final String sinopse;
  final String imagem;
  final DateTime? dataCriacao;
  final DateTime? dataFinalizacao;
  final String editora;
  final int editoraID;
  final List<String> generos;
  final List<String> autores;
  final List<Capitulo> capitulos;

  Manga({
    required this.id,
    required this.titulo,
    required this.sinopse,
    required this.imagem,
    this.dataCriacao,
    this.dataFinalizacao,
    required this.editora,
    required this.editoraID,
    required this.generos,
    required this.autores,
    required this.capitulos,
  });

  Manga.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        titulo = json['titulo'],
        sinopse = json['sinopse'],
        imagem = json['imagem'],
        dataCriacao = DateTime.tryParse(json['data_criacao'].toString()),
        dataFinalizacao = DateTime.tryParse(json['data_finalizacao'].toString()),
        editora = json['editora']?['nome'] ?? '',
        editoraID = json['editora_id'],
        generos = json['generos']?.cast<String>() ?? [],
        autores = json['autores']?.cast<String>() ?? [],
        capitulos = json['capitulos']
                ?.map<Capitulo>((capitulo) => Capitulo.fromJson(capitulo))
                .toList() ??
            [];
}
